packages/webdriverio/src/middlewares.js

```diff
@@ -45,8 +45,33 @@ export const elementErrorHandler = (fn) => (commandName, commandFn) => {
             }).apply(this)
         }
 
-        return fn(commandName, commandFn).apply(this, args)
+        try {
+            return fn(commandName, commandFn).apply(this, args)
+        } catch (error ) {
+            if (error.message.includes(`stale element reference`)) {
+                let currentElement = this;
+
+                // Generate selector array
+                let selectors = [];
+                do {
```

Comment:
let's do a normal while loop here. Even though there is nothing wrong with do/while loops but it seems odd

Reply:
Switched - however, i moved this off to its own command.